core/src/main/scala/io/finch/Input.scala

```diff
@@ -10,6 +13,40 @@ final case class Input(request: Request, path: Seq[String]) {
   def headOption: Option[String] = path.headOption
   def drop(n: Int): Input = copy(path = path.drop(n))
   def isEmpty: Boolean = path.isEmpty
+
+  def withBody(buf: Buf, charset: Option[Charset] = None): Input = {
+    request.content = buf
+    request.contentLength = buf.length.toLong
+    if (charset.isDefined) request.charset = charset.get.displayName()
+    this
+  }
+
+  def withHeaders(headers: (String, String)*): Input = {
+    headers.foreach { case (k, v) => request.headerMap.set(k, v) }
+    this
+  }
+
+  def withForm(params: (String, String)*): Input = {
+    require(request.host.isDefined, "The host has to be defined")
+    require(params.nonEmpty, "Cannot create request without params")
+    val req = RequestBuilder()
+      .addFormElement(params: _*)
+      .addHeaders(request.headerMap.toMap)
+      .url(request.host.get)
+      .buildFormPost()
+    copy(request = req)
```

Comment:
The problem with copying only the body of the request is that headers added by `buildFormPost` won't be preserved.